src/routers/scene_router.py

```diff
@@ -57,18 +57,51 @@ async def get(tick_number: int, player: str = "") -> Dict[str, Union[int, str, L
 async def add(r: AddObjectDTO, background_tasks: BackgroundTasks):
     logger.debug(f"Starting to add object to scene, starting to deserialize object")
     object = Object.deserialize(r.serialized_object)
+    player = state.scene.get_player(object.owner)
+
+    if player is None:
+        logger.warning(f"Player with id: {object.owner} does not exist")
+        raise HTTPException(status_code=400, detail={"message": f"Player with id: {object.owner} does not exist"})
+
+    if player.add_agent_from_object(object) is False:
+        logger.warning(f"Player with id: {object.owner} cannot add agent")
+        cost = player.calculate_slot_cost()
+        raise HTTPException(
```

Comment:
Catch this exception explicitly

Reply:
nope

Reply:
https://fastapi.tiangolo.com/tutorial/handling-errors/